sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/jsonExpressions.scala

```diff
@@ -663,12 +662,7 @@ case class StructsToJson(
     rowSchema, writer, new JSONOptions(options, timeZoneId.get))
 
   @transient
-  lazy val rowSchema = child.dataType match {
-    case st: StructType => st
-    case ArrayType(st: StructType, _) => st
-    case mt: MapType => mt
-    case ArrayType(mt: MapType, _) => mt
-  }
+  lazy val rowSchema = child.dataType
```

Comment:
nit: `rowSchema` -> `intputSchema`. I named this to `rowSchema` because it was always the schema for the row itself. Now, it seems can be other types as well.